### ISSUE TITLE
False positive for Empty Enum Arguments (empty_enum_arguments) on negative boolean functions

### ISSUE THREAD

**Issue by codelobber** [realm/SwiftLint#3927]
---
name: Bug report
about: False positive for Empty Enum Arguments on negative boolean functions
---

### New Issue Checklist

- [X ] Updated SwiftLint to the latest version 0.47
- [ X] I searched for [existing GitHub issues](https://github.com/realm/SwiftLint/issues) 

### Describe the bug

swiftlint trigger rule `empty_enum_arguments` on if statment when it contain negative boolean function return
bug appear only on systems with Xcode 13.3. 13.2.1 work fine.

##### Complete output when running SwiftLint, including the stack trace and command used

```bash
XXX.swift:137:47: warning: Empty Enum Arguments Violation: Arguments can be omitted when matching enums with associated values if they are not used. (empty_enum_arguments)
XXX.swift:98:20: warning: Empty Enum Arguments Violation: Arguments can be omitted when matching enums with associated values if they are not used. (empty_enum_arguments)
XXX.swift:50:20: warning: Empty Enum Arguments Violation: Arguments can be omitted when matching enums with associated values if they are not used. (empty_enum_arguments)
:0:0: error: Warning Threshold Violation: Number of warnings exceeded threshold of 1. (warning_threshold)
```

### Environment

* SwiftLint version: 0.47.0
* Installation method used: CocoaPods
* Configuration file:

```yml
warning_threshold: 1

opt_in_rules:
  - attributes
#  - closure_end_indentation
  - closure_parameter_position
  - closure_spacing
  - collection_alignment
  - colon
  - convenience_type
  - discouraged_object_literal
  - empty_collection_literal
  - empty_count
  - empty_string
  - enum_case_associated_values_count
  - fatal_error_message
  - first_where
  - force_unwrapping
  - identical_operands
  - implicitly_unwrapped_optional #https://google.github.io/swift/#implicitly-unwrapped-optionals
  - joined_default_parameter
  - last_where
  - legacy_random
  - literal_expression_end_indentation
#  - multiline_arguments
#  - multiline_arguments_brackets
  - multiline_function_chains
  - multiline_literal_brackets
  - multiline_parameters
  - multiline_parameters_brackets
  - operator_usage_whitespace
  - optional_enum_case_matching
  - overridden_super_call
  - pattern_matching_keywords
  - prefer_self_type_over_type_of_self
  - reduce_into
  - redundant_nil_coalescing
  - redundant_type_annotation
  - toggle_bool
#  - trailing_closure
  - unneeded_parentheses_in_closure_argument
  - unused_declaration
  - unused_import
  - vertical_whitespace_closing_braces
  - vertical_whitespace_opening_braces
  - yoda_condition

disabled_rules:
  - private_over_fileprivate
  - weak_delegate
  - trailing_comma
  - trailing_whitespace
  - type_name
  - large_tuple
  - block_based_kvo
  - function_parameter_count
  - function_body_length
  - no_fallthrough_only
  - orphaned_doc_comment
  - todo
  - nesting
  - multiple_closures_with_trailing_closure
  - cyclomatic_complexity
  - inclusive_language

identifier_name:
  min_length: 2
  excluded:
    - i
    - j
    - id
    - vc
    - x
    - y


empty_count:
  severity: warning

force_cast: warning
force_try: warning

line_length:
  warning: 110 #https://google.github.io/swift/#general-formatting
  ignores_urls: true
  ignores_function_declarations: true
  ignores_comments: true

file_length: 400
type_body_length: 450

excluded:
  - Pods/
  - Frameworks/
  - XXX/*/.build/
  - XXX/*/Tests

```

* Are you using [nested configurations](https://github.com/realm/SwiftLint#nested-configurations)?
  If so, paste their relative paths and respective contents. - NO
* Which Xcode version are you using (check `xcodebuild -version`)? - Xcode 13.3 

```swift
if !date.isInCurrentYear↓() {
     title += " \(date.year)"
}
```

**Comment by lukeredpath** [comment]
Unsure if this is related but I'm seeing a weird bug with this rule in Xcode 13.3 too. We have a `Clock` type with a `func now` and in my tests I'm calling `self.clock.now()` and SwiftLint reports that this is a violation of `empty_enum_arguments` and keeps auto-correcting it to `self.clock.now`, which does not compile.

**Comment by marcelofabri** [comment]
Are you sure you're using 0.47.0? This sure looks like something we fixed in [0.46.4](https://github.com/realm/SwiftLint/releases/tag/0.46.4) (https://github.com/realm/SwiftLint/issues/3850)

**Comment by sethwhite2** [comment]
I can confirm this is still an issue on 0.47.0. I'm experiencing the same issue.

**Comment by marcelofabri** [comment]
I've double checked and couldn't reproduce it with 0.47.0.

**Comment by EgorVeremeychik** [comment]
See same issue, 0.50.3

**Comment by Jonathanm10** [comment]
Same issue here

Xcode: **15.1**
SwiftLint: **0.53.0**

Error:
```
AddressListView.swift:154:40: error: Empty Enum Arguments Violation: Arguments can be omitted when matching enums with associated values if they are not used. (empty_enum_arguments)
```

Code:
```
154:      if !viewModel.addressLimitReached() {
```
